core/trino-main/src/main/java/io/trino/operator/WorkProcessorSourceOperatorAdapter.java

```diff
@@ -175,6 +177,7 @@ public void close()
             throws Exception
     {
         sourceOperator.close();
+        operatorContext.setLatestMetrics(sourceOperator.getConnectorMetrics());
```

Comment:
Would it better to collect `sourceOperator.getConnectorMetrics()` before `sourceOperator.close()` as sourceOperator could decide to clear it's metrics to release resources on `close` ?

Reply:
Sounds good - moved collections before closing the operator.